src/main/java/org/folio/fqm/utils/SqlFieldIdentificationUtils.java

```diff
@@ -21,13 +20,34 @@ public class SqlFieldIdentificationUtils {
   );
 
   /**
-   * Returns the SQL field corresponding to an entity type column
+   * Returns the version of a SQL field corresponding to an entity type column, intended to be displayed to users
    */
-  public static Field<Object> getSqlField(EntityTypeColumn entityTypeColumn) {
-    if (entityTypeColumn.getValueGetter() == null) {
-      // No value getter is defined for this column. Use column name as the SQL field name.
+  public static Field<Object> getSqlResultsField(EntityTypeColumn entityTypeColumn) {
+    if (entityTypeColumn.getValueGetter() != null) {
+      return getMappedField(entityTypeColumn);
+    }
+    // No value getter is defined for this column. Use column name as the SQL field name.
+    return field(entityTypeColumn.getName());
+  }
```

Comment:
I went ahead and refactored this to reorder the `return`s to better reflect how we pick what to use for a "field" when building a SQL query (so use the value getter if it exists. If not, fall back to the column name). It's kind of a pointless change in this method,  but it made the new `filter` method much clearer and I figured the 2 nearly identical methods should have a similar structure